File: job-worker/src/main/scala/datainsider/jobworker/domain/source/FacebookAdsSource.scala
package datainsider.jobworker.domain.source

import datainsider.jobworker.domain.DataSourceType.DataSourceType
import datainsider.jobworker.domain.Ids.SourceId
import datainsider.jobworker.domain.{DataSource, DataSourceType}

case class FacebookAdsSource(id: SourceId, displayName: String, accessToken: String) extends DataSource {
  override def getId: SourceId = id

  override def getName: String = displayName

  override def getType: DataSourceType = DataSourceType.FacebookAds

  override def getConfig: Map[String, Any] =
    Map(
      "access_token" -> accessToken
    )
}

File: web/di-core/data-ingestion/domain/response/FacebookAdsSource.ts
import { DataSources } from '@core/data-ingestion';
import { DataSource } from '@core/data-ingestion/domain/response/DataSource';
import { SourceId } from '@core/common/domain';

export class FacebookAdsSource implements DataSource {
  readonly className = DataSources.FacebookAds;
  id: SourceId;
  orgId: string;
  displayName: string;
  accessToken: string;
  lastModify: number;

  constructor(id: SourceId, orgId: string, displayName: string, accessToken: string, lastModify: number) {
    this.id = id;
    this.orgId = orgId;
    this.displayName = displayName;
    this.accessToken = accessToken;
    this.lastModify = lastModify;
  }

  static fromObject(obj: any): FacebookAdsSource {
    return new FacebookAdsSource(obj.id, obj.orgId, obj.displayName, obj.accessToken, obj.lastModify);
  }
}
